Automattic/wp-calypso
# WooCommerce: Set defaults during initial setup for currency settings and units

Note: This `wc-api-dev` branch is required to test this branch properly: https://github.com/woocommerce/wc-api-dev/pull/14

This PR hooks into the initial setup and defaults the currency based on the country selected during the setup step. `CAD` for `CA` and `USD` for US. It also sets the default currency position, decimal places, etc. Finally, it sets inches and pounds for our dimensions and weight units, instead of the WooCommerce defaults.

It also takes the general settings batch action/reducer and instead makes it a batch action for all settings, since units are in a different settings category (hence the API requirement).

I left some TODOs for cleanup I would like to do in the future -- but this gets us there for v1.

To Test:
* Go to `https://developer.wordpress.com/docs/api/console/` and set your Calypso preferences to false, so that you can get the set address screen.
* Go to `http://calypso.localhost:3000/store/:site` and enter an address, country, etc.
* Click "Let's Go"
* You can verify units were set correctly by going to "Add Product", and that the currency was set correctly by going to payment settings. You can also use the Redux dev tools to make sure the response came back OK.
* Run `npm run test-client client/extensions/woocommerce/state/` and make sure all tests pass.
